pkg/runtime/runtime.go

```diff
@@ -1513,11 +1531,13 @@ func (a *DaprRuntime) publishMessageHTTP(ctx context.Context, msg *pubsubSubscri
 		// When adding/removing an error here, check if that is also applicable to GRPC since there is a mapping between HTTP and GRPC errors:
 		// https://cloud.google.com/apis/design/errors#handling_errors
 		log.Errorf("non-retriable error returned from app while processing pub/sub event %v: %s. status code returned: %v", cloudEvent[pubsub.IDField], body, statusCode)
+		diag.DefaultComponentMonitoring.PubsubIngressEvent(ctx, msg.pubsub, string(pubsub.Drop), msg.topic, elapsed)
 		return nil
 	}
 
 	// Every error from now on is a retriable error.
 	log.Warnf("retriable error returned from app while processing pub/sub event %v, topic: %v, body: %s. status code returned: %v", cloudEvent[pubsub.IDField], cloudEvent[pubsub.TopicField], body, statusCode)
+	diag.DefaultComponentMonitoring.PubsubIngressEvent(ctx, msg.pubsub, string(pubsub.Retry), msg.topic, elapsed)
```

Comment:
convert `pubsub.Retry`, `pubsub.Drop` etc to be lowercase to be consistent across all diag paths.

Reply:
1. These values are taken from Components Contrib
2. This is not how Golang works. Lowercasing them would mean these fields are unexported, and thus unavailable to `runtime.go`.

Reply:
When I meant lowercase, I meant the values not the variable names.

Reply:
Ah well that makes sense! should have said so :)

Changed